guides/creating_a_new_component.md

```diff
@@ -0,0 +1,386 @@
+        self.assertEqual(iface.process(["#000000"]), ["#000000"])
+
+    def test_static(self):
+        """
+        postprocess
+        """
+        component = gr.ColorPicker("#000000")
+        self.assertEqual(component.get_config().get("value"), "#000000")
+```
+
+## Step 2 - Create a New Svelte Component
+
+Let's see the steps you need to follow to create the frontend of your new component and to map it to its python code:
+- Create a new UI-side Svelte component and figure out where to place it. The alternatives are: create a package for the new component in the [ui/packages folder](https://github.com/gradio-app/gradio/tree/main/ui/packages), if this is completely different from existing components or add the new component to an existing package, such as to the [form package](https://github.com/gradio-app/gradio/tree/main/ui/packages/form). The ColorPicker component for example, was included in the form package because it is similar to components that already exist.
+- Create a file with an appropriate name in the src folder of the package where you placed the Svelte component, note: the name must start with a capital letter. Initially add any text/html to this file so that the component renders something. The Svelte application code for the ColorPicker looks like this:
+
+```typescript
+<script lang="ts">
+	import { createEventDispatcher } from "svelte";
+	import { get_styles } from "@gradio/utils";
+	import { BlockTitle } from "@gradio/atoms";
+	import type { Styles } from "@gradio/utils";
+
+	export let value: string = "#000000";
+	export let style: Styles = {};
+	export let label: string;
+	export let disabled = false;
+	export let show_label: boolean = true;
+
+	$: value;
+	$: handle_change(value);
+
+	const dispatch = createEventDispatcher<{
+		change: string;
+		submit: undefined;
+	}>();
+
+	function handle_change(val: string) {
+		dispatch("change", val);
+	}
+
+	$: ({ classes } = get_styles(style, ["rounded", "border"]));
+</script>
+
+<!-- svelte-ignore a11y-label-has-associated-control -->
+<label class="block">
+	<BlockTitle {show_label}>{label}</BlockTitle>
+	<input
+		type="color"
+		class="gr-box-unrounded {classes}"
+		bind:value
+		{disabled}
+	/>
+</label>
+```
+
+- Export this file inside the index.ts file of the package where you placed the Svelte component by doing `export { default as FileName } from "./FileName.svelte"`. The ColorPicker file is exported in the [index.ts](https://github.com/gradio-app/gradio/blob/main/ui/packages/form/src/index.ts) file and the export is performed by doing: `export { default as ColorPicker } from "./ColorPicker.svelte";`.
+- Create the real component in [ui/packages/app/src/components](https://github.com/gradio-app/gradio/tree/main/ui/packages/app/src/components), copy the folder of another component, rename it and edit the code inside it, keeping the structure. Here you will have three files, the first file is for the Svelte application, and it will look like this:
```

Comment:
```suggestion
- Create the Gradio specific component in [ui/packages/app/src/components](https://github.com/gradio-app/gradio/tree/main/ui/packages/app/src/components), copy the folder of another component, rename it and edit the code inside it, keeping the structure. 

Here you will have three files, the first file is for the Svelte application, and it will look like this:
```